animal.py

```diff
@@ -215,10 +217,36 @@ class Animal(ModelSQL, ModelView, AnimalMixin):
         depends=_DEPENDS_INDIVIDUAL_FIELD)
     active = fields.Boolean('Active')
 
+    lots= fields.One2Many(
+        'stock.lot', 'animal', 'Lots', readonly=True)
+
     # We can't use the 'required' attribute in field because it's
     # checked on view before execute 'create()' function where this
     # field is filled in.
 
+    @classmethod
+    def __register__(cls, module_name):
+        TableHandler = backend.get('TableHandler')
+        table = cls.__table_handler__(module_name)
+        sql_table = cls.__table__()
+        sql_table_animal_lot = 'stock_lot-farm_animal'
+        if not TableHandler.table_exist(sql_table_animal_lot):
+            super(Animal, cls).__register__(module_name)
+            return
+        sql_table_animal_lot = Table(sql_table_animal_lot)
+        update_lot = False
+        if not table.column_exist('lot'):
+            update_lot = True
+        super(Animal, cls).__register__(module_name)
+        table = cls.__table_handler__(module_name)
+        cursor = Transaction().connection.cursor()
+        if update_lot:
+            cursor.execute(*sql_table_animal_lot.select(
+                sql_table_animal_lot.animal, sql_table_animal_lot.lot))
+            for animal_id, lot_id in cursor.fetchall():
+                cursor.execute(*sql_table.update(columns=[sql_table.lot],
+                    values=[lot_id], where=sql_table.id == animal_id))
+
```

Comment:
You can reestructure a bit the code:

It's strange to have to calls to super() even if they're in two different "if" workflows.

Everything becomes much simpler if you declare sql_table_animal_lot when "update_lot" is True (that is, after the call to super()). Also the code becomes more cohesive.

Also cursor declaration should be declared inside the if statement.